src/main/kotlin/com/google/androidstudiopoet/models/PackagesBlueprint.kt

```diff
@@ -16,17 +16,22 @@ limitations under the License.
 
 package com.google.androidstudiopoet.models
 
-data class PackagesBlueprint(private val javaPackageCount: Int,
-                             private val javaClassCount: Int,
-                             private val javaMethodsPerClass: Int,
-                             private val kotlinPackageCount: Int,
-                             private val kotlinClassCount: Int,
-                             private val kotlinMethodsPerClass: Int,
+import com.google.androidstudiopoet.input.CodeConfig
+
+data class PackagesBlueprint(private val javaConfig: CodeConfig?,
+                             private val kotlinConfig: CodeConfig?,
                              val where: String,
                              private val moduleName: String,
                              private val methodsToCallWithin: List<MethodToCall>,
                              val generateTests : Boolean) {
 
+    private val javaPackageCount: Int = javaConfig?.packages ?: 0
+    private val javaClassCount: Int  = javaConfig?.classesPerPackage ?: 0
+    private val javaMethodsPerClass: Int  = javaConfig?.methodsPerClass ?: 0
+    private val kotlinPackageCount: Int  = kotlinConfig?.packages ?: 0
+    private val kotlinClassCount: Int  = kotlinConfig?.classesPerPackage ?: 0
+    private val kotlinMethodsPerClass: Int  = kotlinConfig?.methodsPerClass ?: 0
+
     val javaPackageBlueprints = ArrayList<PackageBlueprint>()
     val kotlinPackageBlueprints = ArrayList<PackageBlueprint>()
     var methodToCallFromOutside: MethodToCall
```

Comment:
@NikitaKozlov this class breaks on the old format, on line 56
 methodToCallFromOutside = if (kotlinPackageCount != 0) {
            kotlinPackageBlueprints.last().methodToCallFromOutside
        } else {
            javaPackageBlueprints.last().methodToCallFromOutside
        }